src/js/legend.js

```diff
@@ -0,0 +1,227 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.defaults("gpii.chartAuthoring.pieChart.legend", {
+        gradeNames: ["gpii.d3ViewComponent", "autoInit"],
+        strings: {
+            legendColHeading:"Legend",
+            labelColHeading:"Label",
+            valueColHeading:"Value"
+        },
+        model: {
+            // dataSet accepts:
+            // 1. an array of objects. Must contain "id", "value" and "label" variables.
+            // Example: [{id: string, value: number, label: string} ... ]
+            dataSet: []
+            // "dataSetWithColors" consolidates the dataset in original order with the array of colors, into a single array of objects
+            // dataSetWithColors: []
+        },
+        legendOptions: {
+            // An array of colors to fill slices generated for corresponding values of model.dataSet
+            // Or, a d3 color scale that's generated based off an array of colors
+            colors: null,
+            sort: true, // Whether or not to sort the data by values when creating the legend
+            showLegendHeadings: true // Whether or not to display column headings in the legend
+        },
+        modelRelay: [{
+            target: "dataSetWithColors",
+            singleTransform: {
+                type: "fluid.transforms.free",
```

Comment:
Rather than using `fluid.transforms.free` you could just make a simple transformation. Maybe one that just adds a property to objects in an array or something.